src/main/java/backend/UpdateController.java

```diff
@@ -146,4 +152,46 @@ private void openRepositoriesInFilters(List<FilterExpression> filterExprs) {
         // No sort or updated, return sort by descending ID, which is the default.
         return Qualifier.getSortComparator(models, "id", true, false);
     }
+
+    /**
+     * Constructs GUIElements (including all necessary references to labels/milestones/users to properly display
+     * the issue) corresponding to a list of issues without changing the order.
+     *
+     * @param models The MultiModel from which necessary references are extracted.
+     * @param filteredAndSortedIssues The list of issues to construct GUIElements for.
+     * @return A list of GUIElements corresponding to the given list of issues.
+     */
+    private List<GuiElement> produceGUIElements(MultiModel models, List<TurboIssue> filteredAndSortedIssues) {
+        return filteredAndSortedIssues.stream().map(issue -> {
+            Optional<Model> modelOfIssue = models.getModelById(issue.getRepoId());
+            assert modelOfIssue.isPresent();
+
+            return new GuiElement(issue,
+                    produceRelevantLabels(modelOfIssue.get(), issue),
+                    models.getMilestoneOfIssue(issue),
+                    models.getAssigneeOfIssue(issue));
+        }).collect(Collectors.toList());
+    }
+
+    /**
+     * Creates a list of relevant labels to an issue, containing current labels of the issue combined with
+     * labels removed from the issue.
+     *
+     * @param model The model containing the relevant issue.
+     * @param issue The issue, the labels of which are to be retrieved.
+     * @return A list of labels needed for displaying the issue properly.
+     */
+    private List<TurboLabel> produceRelevantLabels(Model model, TurboIssue issue) {
+        Stream<TurboLabel> removedLabels = issue.getMetadata().getEvents().stream()
+                .filter(issueEvent -> issueEvent.getType() == IssueEventType.Unlabeled)
+                .map(unlabeledEvent -> {
+                    Optional<TurboLabel> label = model.getLabelByActualName(unlabeledEvent.getLabelName());
+                    assert label.isPresent() : issue + " Label \"" + unlabeledEvent.getLabelName() + "\" not found.";
```

Comment:
Renaming shouldn't be a problem, but deletion is.

I tried to create a new label, assign it to an issue, and then deleted it. This results in the label being automatically removed from the issue. This changes the ETag for labels, and so HT will replace the current list of labels in the model with the updated list, which no longer contains the deleted label. However, the TurboIssueEvents for adding and deleting the label still stays.

Thus, this assertion will trigger. Let's say we remove the assertion and ignore the disappearance of the labels, then in `TurboIssueEvent.display` under the `Unlabeled` case, there is a check if the label is present in the GuiElement, and if no, it will display a new label with the given label name. However, the color will be the default label color (`ffffff`), and this is what happens when `master` is run:

![image](https://cloud.githubusercontent.com/assets/8437638/12271014/0b6aa6e0-b994-11e5-8771-867a5b048189.png)
versus:

![image](https://cloud.githubusercontent.com/assets/8437638/12270997/f6a86cce-b993-11e5-91df-598d1c7b1ee6.png)

The TurboIssueEvent should already contain the label's actual color which is retrievable with the `getLabelColor` method, so we can remove the assert and instead contruct the deleted label from the data in the TurboIssueEvent instead.